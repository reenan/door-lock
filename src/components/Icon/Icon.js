import React, { Component } from 'react'
import PropTypes from 'prop-types'

import './Icon.scss'

class Icon extends Component {
  render() {
    const { icon, onClick, className, size } = this.props

    let style = {}

    if (size) {
      style.fontSize = size + 'px'
    }

    return (
      <span style={style} onClick={onClick} className={`icon ${icon} ${className}`} />
    )
  }
}

Icon.propTypes = {
  icon: PropTypes.string.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func,
  size: PropTypes.number,
}

Icon.defaultProps = {
  className: '',
  size: 0
}

export default Icon
