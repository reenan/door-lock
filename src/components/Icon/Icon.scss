@font-face {
  font-family: "Icons";
  src:
  url("../../resources/icon-font/fa-brands-400.ttf") format("truetype"),
  url("../../resources/icon-font/fa-brands-400.eot") format("eot"),
  url("../../resources/icon-font/fa-brands-400.woff") format("woff"),
  url("../../resources/icon-font/fa-brands-400.woff2") format("woff2"),
  url("../../resources/icon-font/fa-brands-400.svg") format("svg"),
}

@font-face {
  font-family: "Icons";
  src:
  url("../../resources/icon-font/fa-regular-400.ttf") format("truetype"),
  url("../../resources/icon-font/fa-regular-400.eot") format("eot"),
  url("../../resources/icon-font/fa-regular-400.woff") format("woff"),
  url("../../resources/icon-font/fa-regular-400.woff2") format("woff2"),
  url("../../resources/icon-font/fa-regular-400.svg") format("svg"),
}

@font-face {
  font-family: "Icons";
  src:
  url("../../resources/icon-font/fa-solid-900.ttf") format("truetype"),
  url("../../resources/icon-font/fa-solid-900.eot") format("eot"),
  url("../../resources/icon-font/fa-solid-900.woff") format("woff"),
  url("../../resources/icon-font/fa-solid-900.woff2") format("woff2"),
  url("../../resources/icon-font/fa-solid-900.svg") format("svg"),
}

.icon {
  font-family: 'Icons', sans-serif;

  /* Font Awesome uses the Unicode Private Use Area (PUA) to ensure screen
  readers do not read off random characters that represent icons */
  &.lock:before { content: "\f023"; }
  &.lock-open:before { content: "\f3c1"; }
  &.angle-right:before { content: "\f105"; }
  &.check:before { content: "\f00c"; }
  &.close:before { content: "\f00d"; }

}
