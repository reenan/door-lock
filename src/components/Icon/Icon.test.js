import React from 'react'
import Icon from './Icon'

import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const component = renderer.create(
    <Icon icon='Mock' />
  )

  expect(component).toMatchSnapshot();
})

it('renders without crashing with size', () => {
  const component = renderer.create(
    <Icon size={11} icon='Mock' />
  )

  expect(component).toMatchSnapshot();
})
