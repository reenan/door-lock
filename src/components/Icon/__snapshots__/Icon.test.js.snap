// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<span
  className="icon Mock "
  style={Object {}}
/>
`;

exports[`renders without crashing with size 1`] = `
<span
  className="icon Mock "
  style={
    Object {
      "fontSize": "11px",
    }
  }
/>
`;
